### ISSUE TITLE
Crash if in /system/app

### ISSUE THREAD

**Issue by davidcryptic** [luv/mockgeofix#2]
When I move the app from /data/app to /system/app then the app crashes. It's not a permissions issue on the apk (tried 777).

**Comment by luv** [comment]
hi

What android version are you on?

I just tried on android 6.0.1, well cyanogenmod 13.0 to be exact:

```
# mount -o remount,rw /system/
# mv /data/app/github.luv.mockgeofix-1/ /system/app/ 
# rm -rf /data/dalvik-cache/profiles/github.luv.mockgeofix
# reboot
```

Only moving the app (without clearing the cache and rebooting) did result in a crash:
```
 java.lang.RuntimeException: Unable to instantiate application github.luv.mockgeofix.MockGeoFixApp: java.lang.ClassNotFoundException: Didn't find class "github.luv.mockgeofix.MockGeoFixApp" on path: DexPathList[[],nativeLibraryDirectories=[/data/app/github.luv.mockgeofix-1/lib/x86, /vendor/lib, /system/lib]]
                                                                         at android.app.LoadedApk.makeApplication(LoadedApk.java:578)
```

**Comment by davidcryptic** [comment]
Incredibly fast response! Thank you :)

I did not clear the dalvik-cache initially, so I just tried that and rebooted. Still crashing. I'm running Android 4.4.2 on a Samsung GS5.

I just sent the crash report over Google Play so perhaps you are able to see the details?

**Comment by luv** [comment]
You welcome :)

I just tried on Android 4.4.2 (cm 11) and it was actually enough just to move the app to /system/app and reboot the phone and the app is not crashing here.

Strange ...

I have received the crash report in google play:

```
java.io.FileNotFoundException
in android.content.res.AssetManager.openXmlAssetNative
```

but it also says "The error trace you requested is older than the retention period, but similar errors have been recorded recently. These are displayed below. " so the the full stack trace is probably  inaccurate.

However that's a different exception than I was getting when I moved the app and did not reboot.

Maybe some vendor-specific optimization is causing this?

`find /data | grep mockgeofix` on my phone:

```
/data/data/github.luv.mockgeofix
/data/data/github.luv.mockgeofix/lib
/data/data/github.luv.mockgeofix/cache
/data/data/github.luv.mockgeofix/cache/com.android.opengl.shaders_cache
/data/data/github.luv.mockgeofix/shared_prefs
/data/data/github.luv.mockgeofix/shared_prefs/github.luv.mockgeofix_preferences.xml
/data/data/github.luv.mockgeofix/shared_prefs/_has_set_default_values.xml
/data/dalvik-cache/data@app@github.luv.mockgeofix-1.apk@classes.dex
/data/dalvik-cache/system@app@github.luv.mockgeofix-1.apk@classes.dex
```

`find /system | grep mockgeofix` on my phone:
```
/system/app/github.luv.mockgeofix-1.apk
```

Would it make sense to backup the apk, completely wipe out the app from your device and install by "adb push github.luv.mockgeofix-1.apk /system/app/github.luv.mockgeofix-1.apk" instead of installing from google play?

**Comment by davidcryptic** [comment]
Hey man, sorry I did not get back to you sooner. I just saw the email saying you closed the issue (which is fine).

I honestly don't know exactly what went wrong, but the issue persisted through a factory reset, deleting the APK in /system/app and /data/app and deleting the dalvik cache. Here is how I fixed it:

find / | grep mockgeofix
and rm -rf everything that I could find. Then when I reinstalled the app from the apk then it worked flawlessly.

I think you're right that it was something vendor or provider related (AT&T or Samsung, maybe KNOX?). The issue only occurred after I first moved the apk to /system.

Anyways, thanks for the help! You set me on the right path with the "find /system | grep mockgeofix" :)
